core/src/main/java/com/sequenceiq/cloudbreak/service/stackpatch/StackPatchService.java

```diff
@@ -0,0 +1,71 @@
+package com.sequenceiq.cloudbreak.service.stackpatch;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.StackPatch;
+import com.sequenceiq.cloudbreak.domain.stack.StackPatchStatus;
+import com.sequenceiq.cloudbreak.domain.stack.StackPatchType;
+import com.sequenceiq.cloudbreak.repository.StackPatchRepository;
+
+@Service
+public class StackPatchService {
+
+    @Inject
+    private StackPatchRepository stackPatchRepository;
+
+    @Inject
+    private StackPatchUsageReporterService stackPatchUsageReporterService;
+
+    public StackPatch createIfNotExists(Stack stack, StackPatchType stackPatchType) {
+        StackPatch stackPatch = createIfNotExists(stack.getId(), stackPatchType);
+        stackPatch.setStack(stack);
+        return stackPatch;
+    }
+
+    public StackPatch createIfNotExists(Long stackId, StackPatchType stackPatchType) {
```

Comment:
Could you please change the find method to exist and change the JPQL to `exists` instead of the `find`? You can avoid the unnecessary mapping of DB records

Reply:
The StackPatch value is returned, so the mapping is needed